Dockerfile

```diff
@@ -0,0 +1,36 @@
+# This Dockerfile builds btcd from source and creates a small (55 MB) docker container based on alpine linux.
+#
+# Clone this repository and run the following command to build and tag a fresh btcd container:
+#
+# docker build . -t yourregistry/btcd
+#
+# After building the container you can spin it up by running:
+#
+# docker run -it -v btcd-data:/root/.btcd -p 18333:18333 yourregistry/btcd --testnet
+#
+# This command will launch btcd on testnet, exposes the port 18333 and creates a docker named volume btcd-data to store
+# the blockchain and log files.
+#
+# 8333  Mainnet  Bitcoin peer-to-peer port
+# 8334  Mainet RPC port
+# 18333 Testnet Bitcoin peer-to-peer port
+# 18334 Testnet RPC port
+
+FROM golang:1.13.0-alpine3.10 AS build-container
+
+ENV GO111MODULE=on
+
+ADD . /app
+WORKDIR /app
+RUN go install -v . ./cmd/...
+
+
+FROM alpine:3.10
+
+COPY --from=build-container /go/bin /bin
+
+VOLUME ["/root/.btcd"]
+
+EXPOSE 8333 8334 18333 18334
```

Comment:
Regtest & simnet would be super useful too!
Regtest P2P: 18444
Regtest RPC: 18334 (already added though)
Simnet P2P: 18555
Simnet RPC: 18556

Reply:
Sorry, actually regtest RPC (for bitcoin core) is 18443 - the `rpcPort` in `params.go` causes btcd to use 18334 instead of 18443

Reply:
I removed the testnet ports from default expose. It is not necessary to define all possible ports in the docker file. The dockerfile should only contain the minimal port requirements. It is possible to expose any other port in the docker-compose.yml file.

I also added some documentation with examples. I think that is a good start of point.